src/os/FileStore.cc

```diff
@@ -1895,10 +1924,10 @@ void FileStore::_finish_op(OpSequencer *osr)
     o->onreadable_sync->complete(0);
   }
   if (o->onreadable) {
-    op_finisher.queue(o->onreadable);
+    apply_finishers[osr->id % apply_finisher_num]->queue(o->onreadable);
   }
   if (!to_queue.empty()) {
-    op_finisher.queue(to_queue);
+    apply_finishers[osr->id % apply_finisher_num]->queue(to_queue);
```

Comment:
this worries me slightly, since we may reorder the onreadable vs the flush waiters.  In practice, I think the flush watiers are rare.. do these really need to be separate threads?